### ISSUE TITLE
Solidity is not well understood by general people specially lawyers

### ISSUE THREAD

**Issue by ProphetDaniel** [ethereumclassic/README#23]
I think smart contracts should be written somehow like a UML diagram in a visual way stabilishing how it works with easier understanding for specially future lawyers.

I made a code generator that translates UML to C++ language utilizing Acceleo and Papyrus plugins inside eclipse. That is basically a model to text transformation whereas the model is UML and the text is C++ code.

The proposal I make is to utilize a framework like that to translate Ethereum Classic UML based contracts to Solidity language. A model to text transformation powered by Acceleo Engine would be running to compile that in the backstage.

**Comment by whatisgravity** [comment]
I think this would be worth investigating but I fear it may suffer from similar issues to solidity. When you abstract the system to such a degree you may risk introducing issues that become hard to debug since they exist within the language itself, or if it is unable to convey subtitles needed to provide security. That said, I may be wrong and it may lead to greater security. I think it is worth investing time into and if you are interesting in developing a functional requirements list we can leave it as an open project for volunteers to invest time into if they are interested.

**Comment by ProphetDaniel** [comment]
Indeed, this is something that should be tested in a sandbox. The current Ethereum Classic Market capitalization are hundreds of millions of dollars. So there is no margin for introducing unnecessary risks. Although a higher abstraction layer like this avoids attempts to explore solidity language to do evil. A similar approach is being already taken by [Legalese](http://www.coindesk.com/legalese-wins-grant-to-prevent-another-dao-collapse/). Griffith told to Coindesk:

> The fundamental idea is that we wanted to create a programming language for law. Basically it is a mathematical way to describe obligations and relationships.

A layer like that would reduce the complexity of the smart contracts. And the machine would created the Solidity code utilizing best Solidity practices after interpreting the visual model.

**Comment by MrComputerCodes** [comment]
Just throw Solidity away completely and write a domain language. Also, you could get a grant for this from ethdev if you make it a generic EVM project instead of filing it under classic.

**Comment by ProphetDaniel** [comment]
@MrComputerCodes, I think maybe we should open a Consider.it poll to address that question mark. In this regard, the more support we have from community the better I think. So maybe then some Ethereum gurus would weight in thereafter.
**Obs:**
In my opinion DSL's are handy. To develop one I suggest [Xtext](https://www.eclipse.org/Xtext/).

**Comment by whatisgravity** [comment]
Keep in mind there is also serpent and LLL. Solidity is not the only language available it is just the most developed. Providing tools and documentation for others will help users migrate to other possibly safer contract languages.

I'm interested in both this concept and the use of functional languages as alternatives to the existing paradigm. 

Even if the visual language provided only basic but secure functionality to begin with, it would go a long way in allowing for more people to get involved with development. Looking at visual languages like blueprint in UE4 may even be worth researching for additional insight on the topic.

**Comment by EtherTyper** [comment]
:+1: Using UE4 to design smart contracts. Swarm + Whisper + video games = :heart::exclamation:

**Comment by EtherTyper** [comment]
Ethereum will be the GOG.com of the future.

**Comment by ProphetDaniel** [comment]
I took the initiative to put a milestone here. Innovations like that are disruptive and well accepted by the community. I need a solidity expert to volunteer working in this project.

**Comment by ProphetDaniel** [comment]
I found out there's already a plugin available: [UML-solidity](http://marketplace.eclipse.org/content/uml-solidity). The spear blade is getting sharp.
